packages/universal-components/src/RangeDatePicker/DraggableItem.web.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+
+import React from 'react';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import Draggable, { type DraggableData, type Event } from 'react-draggable';
+
+import { designTokens } from '../DesignTokens';
+import { StyleSheet, type StylePropType } from '../PlatformStyleSheet';
+import type { GrabbedSideType } from './RangeDatePickerTypes';
+import type { OnDragEvent } from '../types';
+
+type Props = {
+  +style?: StylePropType,
+  +onDrag: (OnDragEvent, GrabbedSideType) => void,
+  +onDrop: () => void,
+  +grabbedSide: GrabbedSideType,
+};
+
+export default class DraggableItem extends React.Component<Props> {
+  handleDrag = (event: Event, data: DraggableData) => {
+    this.props.onDrag(
+      {
+        nativeEvent: {
+          translationX: data.lastX + data.deltaX,
+          translationY: data.lastY + data.deltaY,
+        },
+      },
+      this.props.grabbedSide,
+    );
+  };
+
+  handleStop = () => {
+    this.props.onDrop();
+  };
+
+  render() {
+    return (
+      <Draggable onDrag={this.handleDrag} onStop={this.handleStop}>
+        <div
+          style={{
+            ...styles.box,
```

Comment:
Do we need to recreate this style on each update? I see that it's mainly because of passed `this.props.style`. But cannot we just use `grabbedSide` prop and handle style internally (have for example some `boxLeft` and `boxRight` style object and use it directly)?

I'm just giving this for consideration, if it will be lot of refactoring I can probably live even with this approach now.

Reply:
OK, done

Reply:
Thanks, definitely better. But I think that we can go even further and get rid of this destructuring and new object creation (basically do it only once during initialisation).

You can for example extract `box` from `styles` and then just spread it into `draggableItemLeft` and `draggableItemRight`. After that you can use `draggableItemLeft/Right` directly.

Reply:
Good idea! Done